VultisigApp/VultisigApp/Services/Thorchain/ThorchainService.swift

```diff
@@ -75,4 +77,49 @@ class ThorchainService {
         
         return cacheEntry.balances
     }
+    
+    func fetchFeePrice() async throws -> UInt64 {
+        let cacheKey = "thorchain-fee-price"
+        if let cachedData: UInt64 = await Utils.getCachedData(cacheKey: cacheKey, cache: cacheFeePrice, timeInSeconds: 60*5) {
+            return cachedData
+        }
+        
+        let urlString = Endpoint.fetchThorchainNetworkInfoNineRealms
+        let data = try await Utils.asyncGetRequest(urlString: urlString, headers: [:])
+        
+        if let result = Utils.extractResultFromJson(fromData: data, path: "native_tx_fee_rune") as? String,
+           let resultNumber = UInt64(result) {
+            self.cacheFeePrice[cacheKey] = (data: resultNumber, timestamp: Date())
+            return resultNumber
+        } else {
+            print("JSON decoding error")
+        }
+        
+        return .zero
+    }
+    
+    private func fetchFeePrice(completion: @escaping (UInt64) -> Void) {
```

Comment:
Can you please remove this function , and also `func fetchFeePrice` below as well 
This is not good design , task in a callback function , and also semaphore..